python/cuml/test/test_benchmark.py

```diff
@@ -67,6 +70,46 @@ def test_run_variations():
     assert (res.n_features == 20).sum() == 2
 
 
+def test_speedup_runner():
+    # Set up data that should deliver accuracy of 0.20 if all goes right
+    class MockAlgo:
+        def __init__(self, t):
+            self.t = t
+
+        def fit(self, X, y):
+            time.sleep(self.t)
+            return
+
+        def predict(self, X):
+            nr = X.shape[0]
+            res = np.zeros(nr)
+            res[0:int(nr / 5.0)] = 1.0
+            return res
+
+    class FastMockAlgo(MockAlgo):
+        def __init__(self):
+            MockAlgo.__init__(self, 0.1)
+
+    class SlowMockAlgo(MockAlgo):
+        def __init__(self):
+            MockAlgo.__init__(self, 1)
+
+    pair = algorithms.AlgorithmPair(
+        SlowMockAlgo,
+        FastMockAlgo,
+        shared_args={},
+        name="Mock",
+        accuracy_function=metrics.accuracy_score,
+    )
+
+    runner = SpeedupComparisonRunner(
+        [20], [5], dataset_name='zeros'
+    )
+    results = runner.run(pair)[0]
+
+    assert results["speedup"] == pytest.approx(10, 0.1)
```

Comment:
Might need to add even more slack around this approx compare so we don't get random failures. I've seen pretty big random timing spikes in benchmarks on CI machines. As long as the fast algo is way faster, we'll be satisfied that we're correct. Maybe 0.4?

Reply:
I increased the slow time to 2 and adjusted the threshold as well.